core-plugin/src/com/google/gct/idea/debugger/CloudBreakpointHandler.java

```diff
@@ -330,13 +349,18 @@ public void registerBreakpoint(@NotNull final XLineBreakpoint<CloudLineBreakpoin
       (CloudLineBreakpointType.CloudLineBreakpoint)cloudIdeBreakpoint;
     if (xIdeBreakpoint.getProperties().isCreatedByServer()) {
       // This newly created IDE breakpoint could be the result of syncing with the server state,
-      // in which case, there is nothing
-      // to do.  Just return.
+      // in which case, there is nothing to do.  Just return.
       // Note that we need to implement this flag as a transient property because this method
       // gets called during construction.
       return;
     }
 
+    if (xIdeBreakpoint.getProperties().isAddedOnServer()) {
```

Comment:
why is a breakpoint that has already been successfully added being passed to registreBreakpoint in the first place?